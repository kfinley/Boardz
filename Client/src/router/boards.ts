import { RouteConfig } from "vue-router";

export const routes: Array<RouteConfig> = [  
  {
    path: "/board/:nameSlug",
    name: "Board",
    component: () =>
      import(/* webpackChunkName: "board" */ "@/components/Board/Board.vue"),
  },
  {
    path: "/board/add",
    name: "AddBoard",
    component: () =>
      import(/* webpackChunkName: "addboards" */ "@/views/AddBoard.vue"),
  },
];
