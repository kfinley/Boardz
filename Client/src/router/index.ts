import VueRouter, { RouteConfig } from "vue-router";

const routes: Array<RouteConfig> = [
  {
    path: "/",
    name: "Home",
    component: () => import(/* webpackChunkName: "boards" */ "@/views/Boards.vue"),
  },
  {
    path: "/about",
    name: "About",
    meta: { allowAnonymous: true },
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () =>
      import(/* webpackChunkName: "about" */ "../views/About.vue"),
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
