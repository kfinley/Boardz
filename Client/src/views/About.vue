<template>
  <div class="about">
    <h1>Really simple boards with user stories as cards.</h1>
  </div>
</template>
