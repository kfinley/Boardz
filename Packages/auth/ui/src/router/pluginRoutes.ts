import { RouteConfig } from "vue-router";
import components from "../components";


const pluginRoutes: Array<RouteConfig> = [
  {
    path: "/auth",
    name: "Auth",
    component: (components as any)["Login"],
    // component: () =>
    //   import(/* webpackChunkName: "auth" */ "../views/Auth.vue"),
  },
];

export default pluginRoutes;
