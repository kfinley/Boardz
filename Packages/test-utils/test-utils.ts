// import { VuexModule } from "vuex-module-decorators";
// import { isEmpty, isEqual, cloneDeep } from "lodash";

// export function stateDiff<STATE>(
//   initialState: STATE,
//   storeModule: VuexModule
// ): Partial<STATE> | null {
//   if (!storeModule.state) {
//     return null;
//   }

//   const objectCopy: any = cloneState<STATE>(storeModule);

//   const result: any = Object.keys(initialState).reduce(
//     (diff: any, key: string) => {
//       if (!objectCopy.hasOwnProperty(key)) {
//         diff[key] = (initialState as any)[key];
//       } else if (isEqual((initialState as any)[key], objectCopy[key])) {
//         delete diff[key];
//       }
//       return diff;
//     },
//     objectCopy
//   );

//   if (isEmpty(result)) {
//     return null;
//   }

//   return result as STATE;
// }

// export function cloneState<STATE>(storeModule: VuexModule): STATE {
//   return Object.keys(storeModule.state!).reduce((state: STATE, key: string) => {
//     // The store must be cast as indexable to get the value.
//     // eslint-disable-next-line @typescript-eslint/no-explicit-any
//     (state as any)[key] = cloneDeep((storeModule as any)[key]); // NOSONAR
//     return state;
//   }, {} as STATE);
// }
