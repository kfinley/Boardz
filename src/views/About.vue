<template>
  <div class="about">
    <h1>Really simple boards with stories as cards.</h1>
  </div>
</template>
