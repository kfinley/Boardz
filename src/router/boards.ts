import { RouteConfig } from "vue-router";

export const routes: Array<RouteConfig> = [
  {
    path: "/",
    name: "Home",
    component: () => import(/* webpackChunkName: "home" */ "@/views/Home.vue"),
  },
  {
    path: "/boards",
    name: "Boards",
    component: () =>
      import(/* webpackChunkName: "boards" */ "@/views/Boards.vue"),
  },
  {
    path: "/auth",
    name: "Auth",
    meta: { allowAnonymous: true },
    component: () =>
      import(/* webpackChunkName: "auth" */ "@/views/Auth.vue"),
  },
];
