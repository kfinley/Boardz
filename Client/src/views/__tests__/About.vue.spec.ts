import Vue from "vue";
import Vuex from "vuex";
import VueRouter from "vue-router";
import { shallowMount } from "@vue/test-utils";

import router from "@/router";
import BoardsPlugin from "@/plugin";
import About from "@/views/About.vue";

Vue.use(Vuex);
Vue.use(VueRouter);

const store = new Vuex.Store({});

Vue.use(BoardsPlugin, {
  router,
  store,
});

describe("About.vue", () => {
  it("About View renders", () => {
    const wrapper = shallowMount(About);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
