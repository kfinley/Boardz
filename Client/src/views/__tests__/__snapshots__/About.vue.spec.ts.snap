// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue About View renders 1`] = `
<div class="about">
  <h1>Really simple boards with user stories as cards.</h1>
</div>
`;
